source/includes/write/transaction.php

```diff
@@ -0,0 +1,54 @@
+<?php
+require 'vendor/autoload.php'; 
+
+$uri = getenv('MONGODB_URI') ?: throw new RuntimeException('Set the MONGODB_URI variable to your connection URI');
+$client = new MongoDB\Client($uri);
+
+// begin-callback
+$receipts = $client->bank->receipts;
+$checking = $client->bank->checking_accounts;
+$saving = $client->bank->saving_accounts;
+
+$accountId = '5678';
+$transferAmount = 1000.00;
+
+$callback = function (MongoDB\Driver\Session $session) 
+    use ($checking, $saving, $receipts, $accountId, $transferAmount): void {
+
+    $checking->updateOne(
+           ['account_id' => $accountId],
+           ['$inc' => ['balance' => -$transferAmount]],
+           ['session' => $session]
+    );
+    
+    $saving->updateOne(
+        ['account_id' => $accountId],
+        ['$inc' => ['balance' => $transferAmount]],
+        ['session' => $session]
+    );
+
+    $summary = sprintf(
+        "SAVINGS +%u CHECKING -%u", $transferAmount, $transferAmount
+    );
+
+    $receipts->insertOne([
+        'account_id' => $accountId,
+        'summary' => $summary,
+        'timestamp' => new MongoDB\BSON\UTCDateTime(),
+    ],
+    ['session' => $session]);
+
+    echo 'Successfully performed transaction!' , PHP_EOL;
+    echo 'Summary: ', $summary, PHP_EOL;
+};
+// end-callback
+
+// begin-txn
+$session = $client->startSession();
+
+try {
+    MongoDB\with_transaction($session, $callback);
+} catch (Exception $e) {
```

Comment:
```suggestion
} catch (MongoDB\Driver\Exception\RuntimeException $e) {
```

Optional suggestion, but the best practice would be to limit the catch to what can be expected at runtime. The general Exception class would cover _anything_ from PHP, including something like an error passing an invalid parameter or option in the driver -- which is better left to blow up in the user's face to alert them to a coding mistake.